src/main/java/com/exactpro/th2/common/schema/factory/CommonFactory.java

```diff
@@ -478,6 +501,113 @@ public static CommonFactory createFromKubernetes(String namespace, String boxNam
+                throw new IllegalStateException("Found several dictionaries");
+            }
+
+            var targetDictionary = dictionaries.iterator().next();
+
+            return new ByteArrayInputStream(getGzipBase64StringDecoder().decode(Files.readString(targetDictionary)));
+        } catch (IOException e) {
+            throw new IllegalStateException("Can not read dictionary", e);
+        }
+    }
+
+    @Override
+    public Set<String> loadDictionaryAliases() {
+        try {
+            Path dictionaryFolder = getPathToDictionaryAliasesDir();
+            LOGGER.debug("Loading dictionaries from folder: {}", dictionaryFolder.toString());
+            Set<String> dictionaries = new HashSet<>();
+            if (Files.exists(dictionaryFolder) && Files.isDirectory(dictionaryFolder)) {
+                try (Stream<Path> files = Files.list(dictionaryFolder)) {
+                    files.filter(Files::isRegularFile).forEach(dictionary -> {
+                        if (Files.isRegularFile(dictionary)) {
+                            dictionaries.add(FilenameUtils.removeExtension(dictionary.getFileName().toString()));
+                        }
+                    });
+                }
+            }
+            return dictionaries;
+        } catch (IOException e) {
+            throw new IllegalStateException("Can not read dictionary", e);
+        }
+    }
+
+    @Override
+    public InputStream loadDictionary(String alias) {
+        var dictionaryName = alias.toLowerCase();
+        try {
+            Path dictionaryFolder = getPathToDictionaryAliasesDir();
+            LOGGER.debug("Loading dictionary alias ({}) from folder: {}", alias, dictionaryFolder);
+            if (Files.exists(dictionaryFolder) && Files.isDirectory(dictionaryFolder)) {
+                try (Stream<Path> stream = Files.list(dictionaryFolder)) {
+                    for (Path pathToAlias : stream.filter(Files::isRegularFile).collect(Collectors.toList())) {
+                        String fileNameWithoutExt = FilenameUtils.removeExtension(pathToAlias.getFileName().toString());
+                        if (fileNameWithoutExt.toLowerCase().equals(dictionaryName)) {
+                            return new ByteArrayInputStream(getGzipBase64StringDecoder().decode(Files.readString(pathToAlias)));
+                        }
+                    }
+                }
+
+            }
+            throw new IllegalStateException("No dictionary found with name-alias '" + alias + "'");
+        } catch (IOException e) {
+            throw new IllegalStateException("Can not read dictionary", e);
+        }
+    }
+
+    @Override
+    public InputStream readDictionary() {
+        return readDictionary(DictionaryType.MAIN);
+    }
+
+    @Override
+    public InputStream readDictionary(DictionaryType dictionaryType) {
+        try {
+            List<Path> dictionaries = null;
+            Path typeFolder = dictionaryType.resolvePathFrom(getPathToDictionaryTypesDir());
```

Comment:
I think this method should also look for dictionary with alias = type for backward compatibility